x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/action/TransportPutInferenceModelAction.java

```diff
@@ -125,6 +130,18 @@ protected void masterOperation(
 
         var requestAsMap = requestToMap(request);
         var resolvedTaskType = ServiceUtils.resolveTaskType(request.getTaskType(), (String) requestAsMap.remove(TaskType.NAME));
+        if (resolvedTaskType == TaskType.EMBEDDING && featureService.clusterHasFeature(state, EMBEDDING_TASK_TYPE) == false) {
```

Comment:
why do we need to do this? is it again only for the inference_usage tests when they run in bwc mode?
then I think we can just check the cluster_feature directly in the yaml test and we don't need this check here.

when we added new task types in the past (e.g. https://github.com/elastic/elasticsearch/pull/119982) we never needed a similar check